src/server/job_func.c

```diff
@@ -773,21 +773,19 @@ job_purge(job *pjob)
 #else	/* not PBS_MOM */
 	if ((pjob->ji_qs.ji_substate != JOB_SUBSTATE_TRANSIN) &&
 		(pjob->ji_qs.ji_substate != JOB_SUBSTATE_TRANSICM)) {
-		if (pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob) {
-			svr_dequejob(pjob);
-			if (pjob->ji_qs.ji_substate == JOB_SUBSTATE_RERUN3)
-				update_subjob_state(pjob, JOB_STATE_QUEUED);
-			else
-				update_subjob_state(pjob, JOB_STATE_EXPIRED);
-		} else {
 			(void)set_entity_ct_sum_queued(pjob, NULL, DECR);
```

Comment:
It looks like the following entity limits are now calculated even for subjobs. Is that what you want?
* max_queued
* max_queued_res
* queued_jobs_thremax_queuedshold
* queued_jobs_threshold_res

Reply:
with rc 4 commit I have edited this change as not to calculate the entity limits for subjobs for this PR.
Is it calculated for the whole array job, we should see in future if we need to consider it for subjobs as well